parlai/tasks/triviaqa/agents.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.dialog_teacher import DialogTeacher
+from .build import build
+
+import json
+import os
+import random
+
+def _path(opt):
+    build(opt)
+
+    return (os.path.join(opt['datapath'], 'TriviaQA', 'qa'),
+            os.path.join(opt['datapath'], 'TriviaQA', 'evidence'))
+
+class DefaultTeacher(DialogTeacher):
+    def __init__(self, opt, shared=None):
+        if opt['datatype'].startswith('train'):
+            self.suffix = 'train'
+        else:
+            self.suffix = 'dev'
+
+        opt['datafile'], self.evidence_dir = _path(opt)
+        self.id = 'triviaqa'
+        super().__init__(opt, shared)
+
+    def setup_data(self, path):
+        datasets = ['web', 'wikipedia']
+
+        for dataset in datasets:
+            dataset_path = os.path.join(path,
+                                        dataset + '-' + self.suffix + '.json')
```

Comment:
Split into different teachers: WikipediaTeacher and WebTeacher. The DefaultTeacher now inherits from MultiTaskTeacher.